electrum/gui/kivy/main_window.py

```diff
@@ -1213,11 +1218,60 @@ def check_pin_code(self, pin):
         if pin != self.electrum_config.get('pin_code'):
             raise InvalidPassword
 
+    def check_password_for_directory(self, old_password, new_password, update=False):
+        dirname = os.path.dirname(self.electrum_config.get_wallet_path())
+        self.logger.info(f'update pw {dirname}')
+        failed = []
+        success = []
+        for filename in os.listdir(dirname):
+            path = os.path.join(dirname, filename)
+            basename = os.path.basename(path)
+            storage = WalletStorage(path)
+            if not storage.is_encrypted():
```

Comment:
what is wallet is already keystore-encrypted (but not storage-encrypted)?

Reply:
this is now handled.